### PR TITLE
/math/fmod && /math/remainder incomplete type bug-fix && /math/min/ fix

### PR DESCRIPTION
**Pull request by jkrude**
The template names are according to other usages in fmod or remainder.

### CODE REVIEW

File: include/alpaka/math/remainder/RemainderHipBuiltIn.hpp [hunk 1]
```diff
@@ -50,20 +50,24 @@ namespace alpaka
...
             //! The standard library remainder trait specialization.
             template<
-                typename TArg>
+                typename Tx,
+                typename Ty>
             struct Remainder<
                 RemainderHipBuiltIn,
-                TArg,
+                Tx,
+                Ty>
```
Comment (tdd11235813) [inline include/alpaka/math/remainder/RemainderHipBuiltIn.hpp]: replace `Ty>` with `Ty,`

File: include/alpaka/math/remainder/RemainderCudaBuiltIn.hpp [hunk 2]
```diff
@@ -37,25 +37,33 @@ namespace alpaka
             using RemainderBase = RemainderCudaBuiltIn;
         };
 
-        namespace traits
+         namespace traits
```
Comment (tdd11235813) [inline include/alpaka/math/remainder/RemainderCudaBuiltIn.hpp]: remove whitespace

**Review by tdd11235813 — DISMISSED** [review]

**Comment by tdd11235813** [comment]
https://github.com/ComputationalRadiationPhysics/alpaka/blob/5d942aa02263ad2c46b200ab6dfbfc64290ef3bb/include/alpaka/math/min/MinHipBuiltIn.hpp#L90

Can you include the fix for the wrong Hip minimum function above (max -> min)? Can be a separate commit, and we can add this to the title of this PR.

**Comment by BenjaminW3** [comment]
After incorporating the review comments you have to rebase this branch onto the current develop branch to get a green build due to #833.

**Review by BenjaminW3 — APPROVED** [review]

**Pull request merged**